zulipterminal/model.py

```diff
@@ -1281,6 +1281,22 @@ def _register_non_subscribed_streams(
             stream["stream_id"]: stream for stream in never_subscribed_streams
         }
 
+    def _get_stream_from_id(self, stream_id: int) -> Union[Subscription, Stream]:
```

Comment:
If we do that, should I cast the `subscribed` and `unsubscribed` streams returned by this method to `Stream`? That might make the usage of this method clearer too, ie. to only access stream properties (and not subscription properties) from the returned stream.

Reply:
A cast shouldn't be necessary once you have the two different functions - you don't have that in the latest version of the code you just pushed?

Reply:
I've casted the stream_dict and the unsubscribed_streams to a `Stream` in the `_get_stream_from_id` in the latest commit, so that methods which use the `_get_stream_from_id` always access only the stream properties and not the subscription properties (which would then warrant the use of `_get_subscription_from_id`).

Reply:
Ah right, but I don't remember seeing an explicit `cast()` to work around mypy - I assume this works in the type system since a Stream is a subset of a Subscription.